src/comodo/drakeSimulator/drakeSimulator.py

```diff
@@ -0,0 +1,243 @@
+        self.duh.add_soft_feet_collisions(plant, xMinMax=xMinMax, yMinMax=yMinMax)
+
+        plant.Finalize()
+        builder.ExportInput(plant.get_actuation_input_port(), "control_input")
+        builder.ExportOutput(plant.get_state_output_port(), "state_output")
+
+        # useful simulator details
+        self.nq = plant.num_positions()
+        self.nv = plant.num_velocities()
+        self.na = plant.num_actuators()
+        self.sim_joint_order = self.duh.get_sim_joint_order(plant, robot_model_sim)
+        # check if the joint ordering is the same
+        # logging.info(
+        #     "Need joint mapping: ", not (sim_joint_order == robot_model.joint_name_list)
+        # )
+
+        # create useful variables
+        self.qpos = np.zeros(self.nq)
+        self.joint_torques = np.zeros(self.na)
+
+        if self.visualize_robot_flag:
+            MeshcatVisualizer.AddToBuilder(builder, scene_graph, self.meshcat)
+            ContactVisualizer.AddToBuilder(
+                builder,
+                plant,
+                self.meshcat,
+                ContactVisualizerParams(
+                    newtons_per_meter=1e3, newton_meters_per_meter=1e1
+                ),
+            )
+
+        diagram = builder.Build()
+        self.simulator = Simulator(diagram)
+
+        # now perform the setup
+        self._setup(s, xyz_rpy)
+
+    def _setup(self, s, xyz_rpy):
+        # similar to DrakeGymEnv we need to expose the required ports
+        system = self.simulator.get_system()
+        self.state_output_port = system.GetOutputPort("state_output")
+        self.control_input_port = system.GetInputPort("control_input")
+
+        self.context = self.simulator.get_mutable_context()
+        self.context.SetTime(0)
+        if not self.simulator.Initialize():
+            logging.error("Drake simulator instance failed to initialize")
+
+        self.simulator.get_system().SetDefaultContext(self.context)
+
+        self.diagram = self.simulator.get_system()
+        plant = self.diagram.GetSubsystemByName("plant")
+        plant_context = self.diagram.GetMutableSubsystemContext(plant, self.context)
+
+        self.set_base_pose_in_drake(xyz_rpy)
+        self.set_joint_vector_in_drake(s)
+        # set initial pose of the robot
+        plant.SetPositions(plant_context, self.qpos)
+        if self.visualize_robot_flag:
+            self.diagram.ForcedPublish(self.context)
+
+    def reset(self):
+        logging.error("Drake simulator reset not implemented, recreate drake_instance")
+        pass
+
+    def set_visualize_robot_flag(self, visualize_robot):
+        # pass meshcat to visualise the robot
+        self.visualize_robot_flag = visualize_robot
+        if self.visualize_robot_flag and not self.active_meshcat:
+            self.meshcat = StartMeshcat()
+            self.active_meshcat = True
+        pass
+
+    # does this need to simulator specific?
+    def set_base_pose_in_drake(self, xyz_rpy):
```

Comment:
why `in_drake` ? We should port this method abstract in the base class, however I do not see the point to call it in drake since it is a method of the drake class

Reply:
I think we can manage this in a separate PR